### PR TITLE
update release flow for lintrunner

### PR DESCRIPTION
**Pull request by PaliC**
Updates the release flow for lintrunner to actually work.

Was tested using https://github.com/pytorch/test-infra/pull/4500

### CODE REVIEW

File: .github/workflows/lintrunner_ci.yml [hunk 1]
```diff
@@ -110,17 +110,13 @@ jobs:
   release:
     name: Release
     runs-on: ubuntu-latest
-    if: "startsWith(github.ref, 'refs/tags/lintrunner/v*')"
```
Comment (huydhn) [inline .github/workflows/lintrunner_ci.yml:113]: I'm curious on why this line is removed, for testing?

File: .github/workflows/lintrunner_ci.yml [hunk 2]
```diff
@@ -110,17 +110,13 @@ jobs:
...
     needs: [ macos, windows, linux, test ]
     steps:
-      - uses: actions/download-artifact@v2
```
Comment (huydhn) [inline .github/workflows/lintrunner_ci.yml:116]: With this removed, what would be the step to download the lintrunner wheel uploaded by the previous build steps

File: .github/workflows/lintrunner_ci.yml (hunk 1 above)
Comment (PaliC) [reply .github/workflows/lintrunner_ci.yml:113]: Oh yess! Thanks for the catch

File: .github/workflows/lintrunner_ci.yml (hunk 2 above)
Comment (PaliC) [reply .github/workflows/lintrunner_ci.yml:116]: Generally the maturin publish command handles all of this for us, so we don't need to download the wheel.

**Pull request merged**